models/gpr.py

```diff
@@ -0,0 +1,84 @@
+from abc import ABC, abstractmethod
+from sklearn.metrics import mean_squared_error
+from sklearn.gaussian_process.kernels import RBF, ConstantKernel, Matern
+from sklearn.gaussian_process import GaussianProcessRegressor
+import pandas as pd
+import numpy as np
+import matplotlib as mpl 
+from matplotlib import pyplot as plt
+from models.model import Model
+class GPR(Model):
+	def __init__(self, dist_array, user_name, dep_name, length, file, lat_n, lon_n):
+		self.dist = dist_array#array-like
+		self.user = file[user_name].values#array-like
+		self.dep = file[dep_name].values#array-like
+		self.depth = []
+		for i in range(len(self.dep)):
+			self.depth.append(self.dep[i] * -1)
+		self.leng = length#int
+		self.gp = GaussianProcessRegressor(normalize_y=True, kernel=Matern()+ConstantKernel(), alpha = 0.0001)
+		self.x_coor = np.linspace(min(self.dist), max(self.dist), self.leng)
+		self.y_coor = np.linspace(min(self.depth), max(self.depth), self.leng)
+		self.xx, self.yy = np.meshgrid(self.x_coor, self.y_coor)
+		self.x = []
+		self.y = []
+		self.xy = []
+		for i in range(len(self.xx)):
```

Comment:
Yeah you might be able to do that, but I think it's more convenient this way because I'm also returning the x and y arrays in the predict() method

Reply:
I just left it as is... it might be slightly more efficient but the way I have it set up, I return the x and y arrays in the predict() method and use them for graphing in the visualization methods, and I don't think it's much more expensive if at all since I need to at least construct one list; therefore, the for loop is necessary anyway